package NPersonalization.NProto;

option go_package = "a.yandex-team.ru/kernel/user_history/proto;userhistory";

enum EEmbeddingType {
    TitleEmbedding = 0;
    QueryEmbedding = 1;
}

enum EModels {
    LogDwelltimeBigrams = 0;
    BoostingXfOneSeAmSsHard = 1;
    BoostingCtr = 2;
    BoostingXfOne = 3;
    BoostingXfOneSE = 4;
    BoostingXfWtd = 5;
    CtrNoMiner = 6;
    LogDtBigramsAMHardQueriesNoClicks = 7;
    LogDtBigramsAMHardQueriesNoClicksMixed = 8;
    MainContentKeywords = 9;
    Multiclick = 10;
    MulticosStreams = 11;
    CtrEngSsHard = 12;
    FpsSpylogAggregatedQueryPart = 13;
    ReformulationsLongestClickLogDt = 14;
    QueryToImageV8Layer = 15; // images
    LogDtBigramsQueryPart = 16;
    ReformulationsQueryEmbedderMini = 17;
}

enum EResultType {
    ORGANIC = 0;
    WIZARD = 1;
    ADV = 2;
}

message TEmbeddingOptions {
    optional EModels Model = 1; // Which DSSM should we use (only if StoreEmbeding is true)
    optional uint64 DwelltimeThreshold = 2; // Set dwelltime threshold
    optional bool LessThanThreshold = 3; // Get clicks with dwelltime larger than threshold if false and less if true
    optional EEmbeddingType EmbeddingType = 4; // Field for chosing between clicks or requests
    optional EResultType ResultType = 5 [default = ORGANIC]; // Filter for choosing between oragnic and wizards
    optional uint64 DwelltimeThresholdFromOppositeSide = 6; // Set optional dwelltime threshold to cut from the opposite side
    optional bool OnlyStateful = 7;
}
